src/components/WorkBuddies/MyWorkBuddy.tsx

```diff
@@ -0,0 +1,122 @@
+import {  useContext, useEffect, useState } from "react";
+import { BuddiesContext } from "../BuddiesContext";
+import classNames from "classnames";
+import "./work-buddies.css";
+import { useReducer } from "react";
+import { Buddy } from "./BuddyInterface";
+import { memo } from "react";
+
+const reducer = (
+  index: number,
+  action: { type: string; contextData?: Buddy[]; myBuddyIndex?: number }
+) => {
+  switch (action.type) {
+    case "next":
+      if (action.contextData) {
+        if (index + 1 < action.contextData.length) {
+          return index + 1;
+        } else {
+          return 0;
+        }
+      } else return 0;
+
+    case "prev":
+      if (action.contextData) {
+        return index - 1 > -1 ? index - 1 : action.contextData.length - 1;
+      } else return 0;
+
+    case "my-buddy":
+      if (action.myBuddyIndex) {
+        return action.myBuddyIndex;
+      } else return 0;
+
+    default:
+      return 1;
+  }
+};
+
+const MyWorkBuddy = memo(({ name }: { name: string }) => {
+  const { contextData } = useContext(BuddiesContext);
+
+  //useReducer
+  const [currentBuddyIndex, dispatch] = useReducer(reducer, 0);
+  const [BuddysName, setBuddysName] = useState<string>();
+  const [myBuddyIndex, setMyBuddyIndex] = useState<number>();
+
+  useEffect(() => {
+    if (name) {
+      setBuddysName(name);
+    }
+  }, [name]);
```

Comment:
Can we simplify this like so:
![image](https://github.com/TheaAleksovDev/ReactConceptsPractice/assets/26555796/05d44655-7bb3-4945-b13b-3e4b80010207)

Reply:
yes, absolutely..